### PR TITLE
Kuanchou/2044 dropdown menu style

### LINKED ISSUE #2044: [GUI UX] variable width of a dropdown menu depending on the context (eg filename)
we have many dropdown menus in the current GUI. There are two width styles observed: one is to fit the width of the displayed context, and the other is a fixed width and is independent to the displayed context. Example is given below. The "data source" dropdown menu on the left is the former while the image dropdown on the top right is the latter. We would need to unify the behavior of the width of a dropdown menu. In addition, the overflow context is displayed differently as the examples at the top right and the bottom right. 

Screen Shot 2022-11-17 at 13 35 21

### PR DESCRIPTION
**Pull request by loveluthien**
**Description**

Closes #2044. 

**Changes**
- Make all dropdown menu widths consistent including the menu in 
-- contour dialog
-- vector dialog
-- fitting dialog
-- PV generator
-- catalog setting
-- moments and fitting panels in spectral profile setting. 
- Make label sizes consistent in tabs of spectral profile setting. 
- Replace labels "Image" with "Data source" in the moments tab of the spectral profile setting and PV generator.
- Some minor UI adjustments. 

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added / new e2e test created~
- [x] ~changelog updated~ / no changelog update needed
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] ~protobuf version bumped~ / protobuf version not bumped
- [x] `BackendService` unchanged / ~`BackendService` changed and corresponding ICD test fix added~

### CODE REVIEW

File: src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss [hunk 1]
```diff
@@ -49,6 +47,31 @@
                 min-width: 2em;
                 width: 3em;
             }
+
+            .bp3-form-content {
+                display: inline-flex;
+                width: calc(100% - 5em);
```
Comment (YuHsuan-Hwang) [inline src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss]: Perhaps add a comment for `5em`.

File: src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss [hunk 2]
```diff
@@ -49,6 +47,31 @@
...
+    .bp3-button:not([class*="bp3-intent-"]) {
+        white-space: nowrap;
+
+        .bp3-button-text {
+            overflow: hidden;
+            text-overflow: ellipsis;
         }
     }
+
+    .file-menu .bp3-button {
```
Comment (YuHsuan-Hwang) [inline src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss]: Can be moved to line 4.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The changes are good. 👍 
Shall we file a new issue for unifying the styles of the dropdown content?

File: src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss (hunk 1 above)
Comment (loveluthien) [reply src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss]: The comment is added.

File: src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss (hunk 2 above)
Comment (loveluthien) [reply src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.scss]: Thanks.

**Comment by loveluthien** [comment]
@YuHsuan-Hwang comments are addressed. I agree to file another issue about the dropdown content and data source infomation (like the figure). 
截圖 2023-12-04 下午4 59 40

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by kswang1029 — CHANGES_REQUESTED** [review]
There are two issues in the contour rendering dialog:
1) the layout of "parameters" should be a two-column layout so that users can see all control UI elements without scrolling down.
Screenshot 2023-12-11 at 13 42 04

2) when a mouse is used (not touch pad), the horizontal scroll bar shows up unnecessarily. It appears there is a small gap somewhere forcing the rendering of the horizontal scroll bar.

https://github.com/CARTAvis/carta-frontend/assets/20819712/f258586d-19ae-48b8-b17d-9accdb7191bc

**Comment by loveluthien** [comment]
@kswang1029 I don't have the second issue in my mac and ubutu. I will relax the width a bit to avoid the horizontal scroll bar.

**Comment by loveluthien** [comment]
@kswang1029 please check the changes.

**Review by kswang1029 — APPROVED** [review]
looks good~

**Pull request merged**